process/transactions/datafield/parser.go

```diff
@@ -0,0 +1,140 @@
+package datafield
+
+import (
+	indexer "github.com/ElrondNetwork/elastic-indexer-go"
+	"github.com/ElrondNetwork/elastic-indexer-go/converters"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/ElrondNetwork/elrond-vm-common/parsers"
+	"math/big"
+)
+
+const (
+	operationRelayedTx                = `relayed`
+	operationTransfer                 = `transfer`
+	minArgumentsQuantityOperationESDT = 2
+	minArgumentsQuantityOperationNFT  = 3
+)
+
+type operationDataFieldParser struct {
+	argsParser         vmcommon.CallArgsParser
+	pubKeyConverter    core.PubkeyConverter
+	shardCoordinator   indexer.ShardCoordinator
+	esdtTransferParser vmcommon.ESDTTransferParser
+}
+
+// NewOperationDataFieldParser will return a new instance of operationDataFieldParser
+func NewOperationDataFieldParser(args *ArgsOperationDataFieldParser) (*operationDataFieldParser, error) {
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, indexer.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.PubKeyConverter) {
+		return nil, indexer.ErrNilPubkeyConverter
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, indexer.ErrNilMarshalizer
+	}
+
+	argsParser := parsers.NewCallArgsParser()
+	esdtTransferParser, err := parsers.NewESDTTransferParser(args.Marshalizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &operationDataFieldParser{
+		argsParser:         argsParser,
+		pubKeyConverter:    args.PubKeyConverter,
+		shardCoordinator:   args.ShardCoordinator,
+		esdtTransferParser: esdtTransferParser,
+	}, nil
+}
+
+// Parse will parse the provided data field
+func (odp *operationDataFieldParser) Parse(dataField []byte, sender, receiver []byte) *ResponseParseData {
+	responseParse := &ResponseParseData{
+		Operation: operationTransfer,
+	}
+
+	function, args, err := odp.argsParser.ParseData(string(dataField))
+	if err != nil {
+		return responseParse
+	}
+
+	switch function {
+	case core.BuiltInFunctionESDTTransfer:
+		return odp.parseESDTTransfer(args, sender, receiver)
```

Comment:
adding error as return also for the private methods

Reply:
no need for error handling in this case.